aws/resource_aws_kinesis_firehose_delivery_stream.go

```diff
@@ -133,6 +133,122 @@ func resourceAwsKinesisFirehoseDeliveryStream() *schema.Resource {
 				},
 			},
 
+			"extended_s3_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"bucket_arn": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"buffer_size": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  5,
+						},
+
+						"buffer_interval": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  300,
+						},
+
+						"compression_format": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "UNCOMPRESSED",
+						},
+
+						"kms_key_arn": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateArn,
+						},
+
+						"role_arn": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"prefix": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"cloudwatch_logging_options": cloudWatchLoggingOptionsSchema(),
+
+						"processing_configuration": {
```

Comment:
I like this. There's a good chance to reuse that structure in the other configurations in the future.